js/src/admin/components/AdminPage.tsx

```diff
@@ -258,7 +268,15 @@ export default abstract class AdminPage<CustomAttrs extends IPageAttrs = IPageAt
       if ((TextareaSettingTypes as readonly string[]).includes(type)) {
         settingElement = <textarea id={inputId} aria-describedby={helpTextId} bidi={this.setting(setting)} {...componentAttrs} />;
       } else {
-        settingElement = <input id={inputId} aria-describedby={helpTextId} type={type} bidi={this.setting(setting)} {...componentAttrs} />;
+        let Tag: any = 'input';
```

Comment:
Is there no way of typing this more specifically?

Reply:
There probably is...  not 100% sure on what it'd be.

Reply:
If you look at the hyperscript signatures under mithril, maybe the type of the first argument to `m` will do?

Would be nice to have, but it's not a blocker.

Reply:
I think I typed this to `string | Component` because almost all JSX elements will be subclasses of our Component class, or strings.

Reply:
Maybe `Mithril.ComponentTypes | string`?

Reply:
@askvortsov1 I think something like that could be helpful in our global typings.

```ts
declare type ElementTag<Attrs = Record<string, unknown>, State = Record<string, unknown>> = string | ComponentTypes<Attrs, State>

Reply:
Sounds good. I suppose we could add that in this PR?

Reply:
```suggestion
        let Tag: VnodeElementTag = 'input';
```